assets/src/edit-story/components/popup/index.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { createPortal } from 'react-dom';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { ADMIN_TOOLBAR_HEIGHT } from '../../constants';
+
+const DEFAULT_WIDTH = 270;
+
+const PopUp = styled.div`
+  position: absolute;
+  z-index: 999;
+  top: ${({ y }) => `${y}px`};
+  left: ${({ x }) => `${x}px`};
+  width: ${({ width }) => width}px;
+`;
+
+function Popup({ root, children, width = DEFAULT_WIDTH }) {
+  const nodeRect = root.getBoundingClientRect();
```

Comment:
This should be done via layout effect. We have to options: either putting the placement params into the state, or just updating inline style manually. Either is ok for portals, imho, b/c this is purely side-effect function. For instance:

```
useLayoutEffect(() => {
  function positionPopup() {
    const anchorRect = anchor.current.getBoundingClientRect();
    const bodyRect = document.body.getBoundingClientRect();
    // ... the math
    popupRef.current.style.width = `...`;
    // or 
    setPopupState({width: ...});
  };
  positionPopup();

  // But you will also need resize handler and ideally also scroll handler.
  window.addEventListener('resize', positionPopup);
  // Scroll handler here does "capture". This will reposition popup if the design panel is scrolled.
  document.addEventListener('scroll', positionPopup, true);
  return () => {
    window.removeEventListener('resize', positionPopup);
    document.removeEventListener('scroll', positionPopup, true);
  };
}, [anchor, width, ...])
```